xdsl/dialects/arm_neon.py

```diff
@@ -111,21 +118,102 @@ def __init__(
         *,
         index: int | None = None,
     ):
+        if isinstance(reg, SSAValue):
+            assert isinstance(reg.type, NEONRegisterType)
+            reg = reg.type
```

Comment:
I can't comment above, but changing 110 to `reg: NEONRegisterType` should do the trick